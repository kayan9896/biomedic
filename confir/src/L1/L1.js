import React, { useState, useEffect, useRef } from 'react';

function L1({tracking, handlLabelClick}) {
    return(
        <>
        <img src={require('./NavBarBg.png')} alt ="NavBarBg" style={{'position':'absolute', top:'962px', left:'0px'}}/>
        <img src={require('./ProgressBarBg.png')} alt ="ProgressBarBg" style={{'position':'absolute', top:'970px', left:'432px'}}/>
        <img src={require('./NavMeasurementsBG.png')} alt ="NavMeasurementsBG" style={{'position':'absolute', top:'970px', left:'960px'}}/>
        
        <img src={require('./Logo.png')} alt ="Logo" style={{'position':'absolute', top:'1041px', left:'13px', zIndex: 12}}/>
        <img src={require('./PatientDataBg.png')} alt ="PatientDataBg" style={{'position':'absolute', top:'980px', left:'13px', zIndex: 12}}/>
        <img src={require('./PatientIcon.png')} alt ="PatientIcon" style={{'position':'absolute', top:'989px', left:'27px', zIndex: 12}}/>

        {tracking?(<img src={require('./APLabel.png')} alt ="APLabel" style={{'position':'absolute', top: '0px', left:'0px', zIndex:1}}/>):(
            <>
            <img src={require('./ManualAPLabelBg.png')} alt ="ManualAPLabelBg" style={{'position':'absolute', top: '0px', left:'0px', zIndex:1}}/>
            <img className="image-button" alt ="ManualAPLabel" src={require('./ManualAPLabel.png')} style={{'position':'absolute', top: '5px', left:'6px', zIndex:10}} onClick={()=>{handlLabelClick('ap')}}/>
            </>
        )}
        {tracking?(<img src={require('./OBLabel.png')} alt ="OBLabel" style={{'position':'absolute', top:'0px', left:'1787px', zIndex:1}}/>):(
            <>
            <img src={require('./ManualOBLabelBg.png')} alt ="ManualOBLabelBg" style={{'position':'absolute', top: '0px', left:'1787px', zIndex:1}}/>
            <img className="image-button" alt ="ManualOBLabel" src={require('./ManualOBLabel.png')} style={{'position':'absolute', top: '5px', left:'1793px', zIndex:10}} onClick={()=>{handlLabelClick('ob')}}/>
            </>
        )}

        </>
    )
}

export default L1;