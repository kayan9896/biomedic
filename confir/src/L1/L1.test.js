// L1.test.js
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom'; // For extended matchers like .toBeInTheDocument()

import L1 from './L1'; // Adjust the path if your component is in a different directory

describe('L1 Component', () => {
  // Mock the handlLabelClick function before each test
  let mockHandlLabelClick;

  beforeEach(() => {
    mockHandlLabelClick = jest.fn(); // Create a fresh mock function for each test
  });

  // --- 1. Test Rendering with Props ---

  describe('when tracking prop is true', () => {
    it('renders the tracking labels and static elements', () => {
      render(<L1 tracking={true} handlLabelClick={mockHandlLabelClick} />);

      // Check for static background images (using alt text or src if available)
      // Since your images don't have alt text, we'll check by src attribute
      expect(screen.getByRole('img', { name: /^NavBarBg$/i })).toBeInTheDocument();
      expect(screen.getByRole('img', { name: /^ProgressBarBg$/i })).toBeInTheDocument();
      expect(screen.getByRole('img', { name: /^NavMeasurementsBG$/i })).toBeInTheDocument();
      expect(screen.getByRole('img', { name: /^Logo$/i })).toBeInTheDocument();
      expect(screen.getByRole('img', { name: /^PatientDataBg$/i })).toBeInTheDocument();
      expect(screen.getByRole('img', { name: /^PatientIcon$/i })).toBeInTheDocument();

      // Check for tracking-specific labels
      const apLabel = screen.getByRole('img', { name: /^APLabel$/i });
      const obLabel = screen.getByRole('img', { name: /^OBLabel$/i });

      expect(apLabel).toBeInTheDocument();
      expect(obLabel).toBeInTheDocument();

      // Assert that manual labels are NOT present
      expect(screen.queryByRole('img', { name: /^ManualAPLabel$/i })).not.toBeInTheDocument();
      expect(screen.queryByRole('img', { name: /^ManualOBLabel$/i })).not.toBeInTheDocument();
    });

    it('matches snapshot when tracking is true', () => {
      const { asFragment } = render(<L1 tracking={true} handlLabelClick={mockHandlLabelClick} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('when tracking prop is false', () => {
    it('renders the manual labels and static elements', () => {
      render(<L1 tracking={false} handlLabelClick={mockHandlLabelClick} />);

      // Check for static background images (same as above)
      expect(screen.getByRole('img', { name: /^NavBarBg$/i })).toBeInTheDocument();
      expect(screen.getByRole('img', { name: /^ProgressBarBg$/i })).toBeInTheDocument();
      expect(screen.getByRole('img', { name: /^NavMeasurementsBG$/i })).toBeInTheDocument();
      expect(screen.getByRole('img', { name: /^Logo$/i })).toBeInTheDocument();
      expect(screen.getByRole('img', { name: /^PatientDataBg$/i })).toBeInTheDocument();
      expect(screen.getByRole('img', { name: /^PatientIcon$/i })).toBeInTheDocument();

      // Check for manual-specific labels and their backgrounds
      const manualApLabelBg = screen.getByRole('img', { name: /^ManualAPLabelBg$/i });
      const manualApLabel = screen.getByRole('img', { name: /^ManualAPLabel$/i });
      const manualObLabelBg = screen.getByRole('img', { name: /^ManualOBLabelBg$/i });
      const manualObLabel = screen.getByRole('img', { name: /^ManualOBLabel$/i });

      expect(manualApLabelBg).toBeInTheDocument();
      expect(manualApLabel).toBeInTheDocument();
      expect(manualObLabelBg).toBeInTheDocument();
      expect(manualObLabel).toBeInTheDocument();

      // Assert that tracking labels are NOT present
      expect(screen.queryByRole('img', { name: /^APLabel$/i })).not.toBeInTheDocument();
      expect(screen.queryByRole('img', { name: /^OBLabel$/i })).not.toBeInTheDocument();
    });

    it('matches snapshot when tracking is false', () => {
      const { asFragment } = render(<L1 tracking={false} handlLabelClick={mockHandlLabelClick} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  // --- 2. Test User Interactions ---

  describe('Interaction when tracking prop is false', () => {
    it('calls handlLabelClick with "ap" when the Manual AP Label is clicked', () => {
      render(<L1 tracking={false} handlLabelClick={mockHandlLabelClick} />);

      const manualApLabel = screen.getByRole('img', { name: /^ManualAPLabel$/i });

      fireEvent.click(manualApLabel);

      expect(mockHandlLabelClick).toHaveBeenCalledTimes(1);
      expect(mockHandlLabelClick).toHaveBeenCalledWith('ap');
    });

    it('calls handlLabelClick with "ob" when the Manual OB Label is clicked', () => {
      render(<L1 tracking={false} handlLabelClick={mockHandlLabelClick} />);

      const manualObLabel = screen.getByRole('img', { name: /^ManualOBLabel$/i });

      fireEvent.click(manualObLabel);

      expect(mockHandlLabelClick).toHaveBeenCalledTimes(1);
      expect(mockHandlLabelClick).toHaveBeenCalledWith('ob');
    });

    it('does not call handlLabelClick when elements without click handlers are clicked', () => {
        // This test ensures that only the intended elements trigger the callback.
        // It implicitly covers that when `tracking` is true, nothing is clickable.
        render(<L1 tracking={false} handlLabelClick={mockHandlLabelClick} />);

        // Click on a background image that doesn't have a click handler
        const navBarBg = screen.getByRole('img', { name: /^NavBarBg$/i });
        fireEvent.click(navBarBg);

        expect(mockHandlLabelClick).not.toHaveBeenCalled();
    });
  });

  describe('Interaction when tracking prop is true (no click interaction expected)', () => {
    it('does not call handlLabelClick when tracking labels are present', () => {
      render(<L1 tracking={true} handlLabelClick={mockHandlLabelClick} />);

      // Attempt to click the APLabel and OBLabel, which should not have click handlers
      const apLabel = screen.getByRole('img', { name: /^APLabel$/i });
      const obLabel = screen.getByRole('img', { name: /^OBLabel$/i });

      // Note: fireEvent.click will still trigger if an element is clicked,
      // but the `handlLabelClick` will only be called if there's an `onClick` prop
      // on that specific element. In this case, there isn't, so the mock shouldn't be called.
      fireEvent.click(apLabel);
      fireEvent.click(obLabel);

      expect(mockHandlLabelClick).not.toHaveBeenCalled();
    });
  });
});