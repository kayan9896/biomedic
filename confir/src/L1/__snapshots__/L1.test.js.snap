// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`L1 Component when tracking prop is false matches snapshot when tracking is false 1`] = `
<DocumentFragment>
  <img
    alt="NavBarBg"
    src="mock-image-path"
    style="position: absolute; top: 962px; left: 0px;"
  />
  <img
    alt="ProgressBarBg"
    src="mock-image-path"
    style="position: absolute; top: 970px; left: 432px;"
  />
  <img
    alt="NavMeasurementsBG"
    src="mock-image-path"
    style="position: absolute; top: 970px; left: 960px;"
  />
  <img
    alt="Logo"
    src="mock-image-path"
    style="position: absolute; top: 1041px; left: 13px; z-index: 12;"
  />
  <img
    alt="PatientDataBg"
    src="mock-image-path"
    style="position: absolute; top: 980px; left: 13px; z-index: 12;"
  />
  <img
    alt="PatientIcon"
    src="mock-image-path"
    style="position: absolute; top: 989px; left: 27px; z-index: 12;"
  />
  <img
    alt="ManualAPLabelBg"
    src="mock-image-path"
    style="position: absolute; top: 0px; left: 0px; z-index: 1;"
  />
  <img
    alt="ManualAPLabel"
    class="image-button"
    src="mock-image-path"
    style="position: absolute; top: 5px; left: 6px; z-index: 10;"
  />
  <img
    alt="ManualOBLabelBg"
    src="mock-image-path"
    style="position: absolute; top: 0px; left: 1787px; z-index: 1;"
  />
  <img
    alt="ManualOBLabel"
    class="image-button"
    src="mock-image-path"
    style="position: absolute; top: 5px; left: 1793px; z-index: 10;"
  />
</DocumentFragment>
`;

exports[`L1 Component when tracking prop is true matches snapshot when tracking is true 1`] = `
<DocumentFragment>
  <img
    alt="NavBarBg"
    src="mock-image-path"
    style="position: absolute; top: 962px; left: 0px;"
  />
  <img
    alt="ProgressBarBg"
    src="mock-image-path"
    style="position: absolute; top: 970px; left: 432px;"
  />
  <img
    alt="NavMeasurementsBG"
    src="mock-image-path"
    style="position: absolute; top: 970px; left: 960px;"
  />
  <img
    alt="Logo"
    src="mock-image-path"
    style="position: absolute; top: 1041px; left: 13px; z-index: 12;"
  />
  <img
    alt="PatientDataBg"
    src="mock-image-path"
    style="position: absolute; top: 980px; left: 13px; z-index: 12;"
  />
  <img
    alt="PatientIcon"
    src="mock-image-path"
    style="position: absolute; top: 989px; left: 27px; z-index: 12;"
  />
  <img
    alt="APLabel"
    src="mock-image-path"
    style="position: absolute; top: 0px; left: 0px; z-index: 1;"
  />
  <img
    alt="OBLabel"
    src="mock-image-path"
    style="position: absolute; top: 0px; left: 1787px; z-index: 1;"
  />
</DocumentFragment>
`;
